woocommerce/woocommerce-ios
# [Analytics Hub] Exit Analytics Hub and display an error if date range selection fails

Part of: #8213

## Description

When selecting a date range fails, we now pop out of the Analytics Hub back to the My Store view and display a notice about the error.

We expect this should only happen in rare cases, when `AnalyticsHubTimeRangeSelection` fails to unwrap the current and previous time ranges.

## Changes

* In the analytics hub view model, we now catch `AnalyticsHubTimeRangeSelection.TimeRangeGeneratorError` errors. We set an error notice and set `errorSelectingTimeRange` to true.
* In the analytics hub view, we use the system notice presenter to enqueue the error notice when the hosting controller disappears, and pop the hosting controller when `errorSelectingTimeRange` is true.

## Testing

You can force a failed date range selection by temporarily editing one of the methods that unwraps the time range to always throw an error:

https://github.com/woocommerce/woocommerce-ios/blob/0af536a4dacda18538078cb996a298a98afde553/WooCommerce/Classes/ViewRelated/Dashboard/Analytics%20Hub/Time%20Range/AnalyticsHubTimeRangeSelection.swift#L68-L73

Then:

1. Build and run the app.
2. Tap the "See more" button.
3. Confirm that you are popped out of the Analytics Hub and an error notice appears on the My Store screen.

## Screenshots

https://user-images.githubusercontent.com/8658164/205310974-8721ae3c-86b3-4f08-81c0-d895141b6816.mp4




## Submitter Checklist

Update release notes:
